amqp/src/main/java/io/cloudevents/amqp/ProtonBasedAmqpMessageFactory.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2018-Present The CloudEvents Authors
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.cloudevents.amqp;
+
+
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.apache.qpid.proton.amqp.messaging.ApplicationProperties;
+import org.apache.qpid.proton.message.Message;
+
+import io.cloudevents.amqp.impl.ProtonBasedAmqpBinaryMessageReader;
+import io.cloudevents.amqp.impl.ProtonBasedAmqpMessageWriter;
+import io.cloudevents.amqp.impl.AmqpConstants;
+import io.cloudevents.core.message.MessageReader;
+import io.cloudevents.core.message.MessageWriter;
+import io.cloudevents.core.message.impl.GenericStructuredMessageReader;
+import io.cloudevents.core.message.impl.MessageUtils;
+import io.cloudevents.core.message.impl.UnknownEncodingMessageReader;
+import io.cloudevents.rw.CloudEventWriter;
+/**
+ * A factory class providing convenience methods for creating MessageReader and MessageWriter instances based on
+ * {@link <a href="https://github.com/apache/qpid-proton-j">Qpid Proton</a>}.
+ */
+@ParametersAreNonnullByDefault
+public final class ProtonBasedAmqpMessageFactory {
```

Comment:
I think without Based is fine, let's not be too much verbose :smile: Same for other classes

Reply:
sure 😄